### PR TITLE
feat(cli): Add CBOR and JSON schemaless decoding for `cat --json`

### PR DESCRIPTION
**Pull request by jon-chuang**
### Public-Facing Changes
`mcap cat --json` allows for schemaless json and cbor encodings

### Description
`mcap cat --json` allows for schemaless json and cbor encodings

Depends on: 
- https://github.com/foxglove/mcap/pull/857

https://github.com/foxglove/mcap/issues/763
https://github.com/foxglove/mcap/issues/847

### CODE REVIEW

File: python/mcap/tests/test_writer.py [hunk 1]
```diff
@@ -115,3 +117,34 @@ def test_out_of_order_messages():
     chunk_index = next(r for r in records if isinstance(r, ChunkIndex))
     assert chunk_index.message_start_time == 0
     assert chunk_index.message_end_time == 100
+
+@pytest.mark.parametrize(
+    "null_schema", [(True,), (False,)]
+)
+def test_generate_sample_cbor_data(null_schema: bool):
```
Comment (jon-chuang) [inline python/mcap/tests/test_writer.py:124]: This is the wrong place for this test, but we should test this functionality somewhere since it was also the source of a SIGSEV...

**Comment by jon-chuang** [comment]
The CLA portal is currently giving a blank page: https://cla-assistant.io/foxglove/mcap?pullRequest=848

File: go/cli/mcap/cmd/cat.go [hunk 2]
```diff
@@ -109,57 +110,80 @@ func printMessages(
 			}
 			continue
 		}
-		switch schema.Encoding {
-		case "ros1msg":
-			transcoder, ok := transcoders[channel.SchemaID]
-			if !ok {
-				packageName := strings.Split(schema.Name, "/")[0]
-				transcoder, err = ros.NewJSONTranscoder(packageName, schema.Data)
+		if schema == nil || schema.Encoding == "" {
```
Comment (jtbandes) [inline go/cli/mcap/cmd/cat.go:113]: Thanks for fixing the bug – please make a separate PR for this though so it can be landed independently of the discussion around CBOR.

Comment (jon-chuang) [reply go/cli/mcap/cmd/cat.go:113]: merged https://github.com/foxglove/mcap/pull/857

File: python/mcap/tests/test_writer.py (hunk 1 above)
Comment (jon-chuang) [reply python/mcap/tests/test_writer.py]: Where should we put this test?

**Comment by jtbandes** [comment]
Closing this PR since we haven't seen much community interest in CBOR, and we've identified that the first step needs to be an update to the spec / well-known types registry (see https://github.com/foxglove/studio/pull/5574#issuecomment-1563425843 & https://github.com/foxglove/studio/pull/5574#issuecomment-1563482157). Thanks for the contribution!
